File: Estado.java
import java.util.HashSet;

class Estado {
    static int generalId = 1;
    HashSet<Transicion> transiciones;
    int id;
    int token;
    boolean aceptacion;

    //=========================

    int getId(){return this.id;}
    int getToken(){return this.token;}
    boolean getAceptacion(){return this.aceptacion;}
    HashSet<Transicion> getTransiciones(){return this.transiciones;}

    void setAceptacion(boolean toAceptacion){this.aceptacion = toAceptacion;}
    void setToken(int toToken){this.token = toToken;}
    void setId(int toId){ this.id = toId; }
    
    void addTransicion(char toChar, Estado toEstado){this.transiciones.add(new Transicion(toChar, toEstado));}

    Estado(){
        this.token = -1;
        this.id = generalId;
        this.aceptacion = false;
        this.transiciones = new HashSet<>();
        this.transiciones.clear();
        generalId++;
    }
}

File: Estado.cs
using System.Collections.Generic;

namespace LexicAnalyzer
{
    class Estado
    {
        static int generalId = 0;
        int id;
        bool aceptacion;
        int token;
        Dictionary<char, Estado> transiciones;

        //===========================

        public void setAceptacion(bool toSet) { this.aceptacion = toSet; }
        public void setToken(int toSet) { this.token = toSet; }
        public void addTransicion(char toChar, Estado toEstado) { transiciones.Add(toChar, toEstado); }

        public int getId() { return id; }
        public bool getAceptacion() { return aceptacion; }
        public Dictionary<char, Estado> getTransiciones() { return transiciones; }

        public Estado()
        {
            this.id = generalId;
            generalId++;
            this.aceptacion = false;
            this.transiciones = new Dictionary<char, Estado>();
            this.transiciones.Clear();
        }
    }
}
